### PR TITLE
fix: fix region re-open bug

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

fix region re-opening bug

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
#2407

### CODE REVIEW

File: src/datanode/src/datanode.rs [hunk 1]
```diff
@@ -246,7 +246,8 @@ impl DatanodeBuilder {
 
         info!("going to open {} regions", regions.len());
 
-        for (region_id, engine, region_dir) in regions {
+        for (region_id, engine, path) in regions {
```
Comment (killme2008) [inline src/datanode/src/datanode.rs:249]: Great catch! It's better to add a test to cover this case.

Comment (WenyXu) [reply src/datanode/src/datanode.rs:249]: Let's leave this as homework to @waynexia 😚

**Comment by fengjiachun** [comment]
Is this a duplicate of #2409?

**Comment by WenyXu** [comment]
> Is this a duplicate of #2409?

yep🥲

**Pull request closed without merging**